pkg/cmd/run/rerun/rerun.go

```diff
@@ -39,8 +40,19 @@ func NewCmdRerun(f *cmdutil.Factory, runF func(*RerunOptions) error) *cobra.Comm
 
 	cmd := &cobra.Command{
 		Use:   "rerun [<run-id>]",
-		Short: "Rerun a failed run",
-		Args:  cobra.MaximumNArgs(1),
+		Short: "Rerun a run",
+		Long: heredoc.Docf(`
+			Rerun an entire run, only failed jobs, or a specific job from a run.
+
+			Note that due for historical reasons, the %[1]s--job%[1]s flag may not take what you expect.
+			Specifically, when navigating to a job in the browser, the URL looks like this:
+			%[1]shttps://github.com/<org>/<repo>/actions/runs/<run-id>/jobs/<job-id>%[1]s.
+
+			However, this %[1]sjob-id%[1]s should not be used with the %[1]s--job%[1]s flag and will result in the
+			API returning %[1]s404 NOT FOUND%[1]s. Instead, you can get the correct job IDs using the following command:
+			%[1]sgh run view <run-id> --json jobs --jq '.jobs[] | {name, databaseId}'%[1]s.
```

Comment:
This invocation could be made shorter for clarity. If the goal is to demonstrate programmatic access to job IDs, then `gh run view <run-id> --json jobs` would suffice (with extracting the `databaseId` being left as an exercise to the reader). If the goal could be satisfied by seeing job IDs in human-readable format, then note that `gh run view <run-id>` prints them as well.

Reply:
I'm happy to change this if you feel strongly but I think it needs to be explained that the field called `databaseId` is the one the user should be looking for. `databaseId` isn't a concept that exists in the API and doesn't link to being a `job-id`. Having it in the command seems as useful as a wordy explanation to me.